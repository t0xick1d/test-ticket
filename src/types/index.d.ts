// declare module '*.module.scss' {
//    const classes: { [key: string]: string };
//    const content: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
//    export default classes;
// }
