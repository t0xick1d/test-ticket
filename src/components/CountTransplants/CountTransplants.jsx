import React from 'react';
import FormGroup from '@mui/material/FormGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';

import style from './styled.module.scss';

const CountTransplants = ({ transplants, setTransplants }) => {
   return (
      <div className={style.container}>
         <FormGroup>
            <FormControlLabel
               control={<Checkbox checked={transplants.all} onChange={setTransplants} name="all" />}
               label="Все"
            />
            <FormControlLabel
               control={
                  <Checkbox
                     checked={transplants.noneTransplants}
                     onChange={setTransplants}
                     name="noneTransplants"
                  />
               }
               label="Без пересадок"
            />
            <FormControlLabel
               control={
                  <Checkbox
                     checked={transplants.oneTransplants}
                     onChange={setTransplants}
                     name="oneTransplants"
                  />
               }
               label="1 пересадка"
            />
            <FormControlLabel
               control={
                  <Checkbox
                     checked={transplants.twoTransplants}
                     onChange={setTransplants}
                     name="twoTransplants"
                  />
               }
               label="2 пересадки"
            />
            <FormControlLabel
               control={
                  <Checkbox
                     checked={transplants.threeTransplants}
                     onChange={setTransplants}
                     name="threeTransplants"
                  />
               }
               label="3 пересадки"
            />
         </FormGroup>
      </div>
   );
};

export default CountTransplants;
