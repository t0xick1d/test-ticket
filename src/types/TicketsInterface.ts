export interface TicketsI {
   price: number;
   carrier: number;
   date: string;
   segments: [SegmentsI];
}
export interface SegmentsI {
   origin: string;
   destination: string;
   departureTime: string;
   duration: number;
   stops: string[];
}
