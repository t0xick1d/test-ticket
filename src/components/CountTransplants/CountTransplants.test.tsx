// import React from 'react';
// import { render } from '@testing-library/react';
// import CountTransplants from './CountTransplants';
// import { TransplantsI } from '../../types/TicketsInterface';

// describe('CountTransplants snapshot', () => {
//    const mockTransplants: TransplantsI = {
//       all: false,
//       noneTransplants: false,
//       oneTransplants: false,
//       twoTransplants: false,
//       threeTransplants: false,
//    };

//    const mockSetTransplants = jest.fn();

//    it('matches snapshot', () => {
//       const { asFragment } = render(
//          <CountTransplants transplants={mockTransplants} setTransplants={mockSetTransplants} />,
//       );
//       expect(asFragment()).toMatchSnapshot();
//    });
// });

import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import CountTransplants from './CountTransplants';
import { TransplantsI } from '../../types/TicketsInterface';

describe('CountTransplants component', () => {
   const mockTransplants: TransplantsI = {
      all: false,
      noneTransplants: false,
      oneTransplants: false,
      twoTransplants: false,
      threeTransplants: false,
   };

   const mockSetTransplants = jest.fn();

   it('renders all checkboxes with correct labels', () => {
      render(
         <CountTransplants transplants={mockTransplants} setTransplants={mockSetTransplants} />,
      );
      expect(screen.getByLabelText('Все')).toBeInTheDocument();
      expect(screen.getByLabelText('Без пересадок')).toBeInTheDocument();
      expect(screen.getByLabelText('1 пересадка')).toBeInTheDocument();
      expect(screen.getByLabelText('2 пересадки')).toBeInTheDocument();
      expect(screen.getByLabelText('3 пересадки')).toBeInTheDocument();
   });

   it('calls setTransplants on checkbox change', () => {
      render(
         <CountTransplants transplants={mockTransplants} setTransplants={mockSetTransplants} />,
      );
      const checkbox = screen.getByTestId('checkbox-all');
      fireEvent.click(checkbox);
      expect(mockSetTransplants).toHaveBeenCalledTimes(1);
   });

   it('matches snapshot', () => {
      const { asFragment } = render(
         <CountTransplants transplants={mockTransplants} setTransplants={mockSetTransplants} />,
      );
      expect(asFragment()).toMatchSnapshot();
   });
});
