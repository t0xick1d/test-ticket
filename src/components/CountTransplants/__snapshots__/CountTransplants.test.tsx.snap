// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountTransplants component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          data-testid="checkbox-all"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            data-indeterminate="false"
            name="all"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          Все
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          data-testid="checkbox-none"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            data-indeterminate="false"
            name="noneTransplants"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          Без пересадок
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          data-testid="checkbox-one"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            data-indeterminate="false"
            name="oneTransplants"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          1 пересадка
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          data-testid="checkbox-two"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            data-indeterminate="false"
            name="twoTransplants"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          2 пересадки
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          data-testid="checkbox-three"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            data-indeterminate="false"
            name="threeTransplants"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          3 пересадки
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CountTransplants snapshot matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          data-testid="checkbox-all"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            data-indeterminate="false"
            name="all"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          Все
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          data-testid="checkbox-none"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            data-indeterminate="false"
            name="noneTransplants"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          Без пересадок
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          data-testid="checkbox-one"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            data-indeterminate="false"
            name="oneTransplants"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          1 пересадка
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          data-testid="checkbox-two"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            data-indeterminate="false"
            name="twoTransplants"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          2 пересадки
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          data-testid="checkbox-three"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            data-indeterminate="false"
            name="threeTransplants"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          3 пересадки
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
